src/integrators/VL_3D_cuda.cu

```diff
@@ -166,29 +225,46 @@ void VL_Algorithm_3D_CUDA(Real *d_conserved, Real *d_grav_potential, int nx, int
     hipLaunchKernelGGL(Calculate_HLL_Fluxes_CUDA, dim1dGrid, dim1dBlock, 0, 0, Q_Ly, Q_Ry, F_y, nx, ny, nz, n_ghost, gama, 1, n_fields);
     hipLaunchKernelGGL(Calculate_HLL_Fluxes_CUDA, dim1dGrid, dim1dBlock, 0, 0, Q_Lz, Q_Rz, F_z, nx, ny, nz, n_ghost, gama, 2, n_fields);
     #endif //HLLC
+    #ifdef HLLD
+    hipLaunchKernelGGL(mhd::Calculate_HLLD_Fluxes_CUDA, dim1dGrid, dim1dBlock, 0, 0, Q_Lx, Q_Rx, &(dev_conserved_half[(grid_enum::magnetic_x) * n_cells]), F_x, nx, ny, nz, n_ghost, gama, 0, n_fields);
+    hipLaunchKernelGGL(mhd::Calculate_HLLD_Fluxes_CUDA, dim1dGrid, dim1dBlock, 0, 0, Q_Ly, Q_Ry, &(dev_conserved_half[(grid_enum::magnetic_y) * n_cells]), F_y, nx, ny, nz, n_ghost, gama, 1, n_fields);
+    hipLaunchKernelGGL(mhd::Calculate_HLLD_Fluxes_CUDA, dim1dGrid, dim1dBlock, 0, 0, Q_Lz, Q_Rz, &(dev_conserved_half[(grid_enum::magnetic_z) * n_cells]), F_z, nx, ny, nz, n_ghost, gama, 2, n_fields);
+    #endif //HLLD
     CudaCheckError();
 
     #ifdef DE
     // Compute the divergence of Vel before updating the conserved array, this solves synchronization issues when adding this term on Update_Conserved_Variables_3D
     hipLaunchKernelGGL(Partial_Update_Advected_Internal_Energy_3D, dim1dGrid, dim1dBlock, 0, 0,  dev_conserved, Q_Lx, Q_Rx, Q_Ly, Q_Ry, Q_Lz, Q_Rz, nx, ny, nz, n_ghost, dx, dy, dz,  dt, gama, n_fields );
     CudaCheckError();
-    #endif
+    #endif  //DE
 
+    #ifdef  MHD
+    // Step 5.5: Compute the Constrained transport electric fields
+    hipLaunchKernelGGL(mhd::Calculate_CT_Electric_Fields, dim1dGrid, dim1dBlock, 0, 0, F_x, F_y, F_z, dev_conserved_half, ctElectricFields, nx, ny, nz, n_cells);
+    CudaCheckError();
+    #endif  //MHD
 
     // Step 6: Update the conserved variable array
     hipLaunchKernelGGL(Update_Conserved_Variables_3D, dim1dGrid, dim1dBlock, 0, 0, dev_conserved, Q_Lx, Q_Rx, Q_Ly, Q_Ry, Q_Lz, Q_Rz, F_x, F_y, F_z, nx, ny, nz, x_off, y_off, z_off, n_ghost, dx, dy, dz, xbound, ybound, zbound, dt, gama, n_fields, density_floor, dev_grav_potential);
     CudaCheckError();
 
+    #ifdef  MHD
```

Comment:
Does the MHD update care about the conserved variable update at all? If so, it should be after the dual energy update. If not, it doesn't matter. I'm just not 100% sure about the ordering here.

Reply:
It doesn't. It only uses the CT electric fields and previous magnetic fields then writes out the updated fields.

Reply:
The two different updates should be entirely independent and could be run in separate streams